WI #616: Addition of five columns related to `Pricing Currency` (#848)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/columns/pricingcurrencylistunitprice.md

# Pricing Currency List Unit Price

The Pricing Currency List Unit Price represents the suggested provider-published unit price for a single [Pricing Unit](#pricingunit) of the associated SKU, exclusive of any discounts. This price is denominated in the [Pricing Currency](#pricingcurrency). The Pricing Currency List Unit Price is commonly used for calculating savings based on various rate optimization activities.

The PricingCurrencyListUnitPrice column adheres to the following requirements:

> * PricingCurrencyListUnitPrice MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when the provider supports pricing and billing in different currencies.

Reviewer: Could someone justify why this is a MUST? In my experience, some vendors do not want to show List Price.

Author: @gparker-at-sf Could you provide some context as to why vendors might not want to show List Price?  Certainly, a vendor could sustain a conformance gap if they don't want to provide it.  

Could you say why, for the practitioner, it would be better to be a SHOULD or MAY?

Author: @gparker-at-sf 
Let's change gears, because I hope we can easily resolve this, based on something I missed in the draft.

Consider the requirement for presence of `ListUnitPrice`:

> The ListUnitPrice column MUST be present in a FOCUS dataset when the provider publishes unit prices exclusive of discounts.

If we added this same requirement here on `PricingCurrencyListUnitPrice`, then would that check the box for you?  I suspect it would.  I had meant to have the presence requirement be in sync between these two columns, but clearly I missed that.

```suggestion
* PricingCurrencyListUnitPrice MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when: 
  * The provider supports pricing and billing in different currencies.
  * The provider publishes unit prices exclusive of discounts. 
```

@ijurica I wasn't immediately sure how we're supposed to express AND conditions in normative requirements, based on the newly-approved guidelines.  Could you confirm that this structure looks correct?

Reviewer: Propose that we clarify that there is an 'AND' on those conditions and not an 'OR'

Author: @gparker-at-sf Yes, definitely need to express the "AND" aspect of this -- thus my question to Irena above.  Will wait to hear from her, and then we can resolve!

> * PricingCurrencyListUnitPrice MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when the provider supports pricing and billing in different currencies and publishes unit prices exclusive of discounts.

Author: A revision for the group to consider:

```suggestion
* PricingCurrencyListUnitPrice presence in a [*FOCUS dataset*](#glossary:FOCUS-dataset) is defined as follows:
  * PricingCurrencyListUnitPrice MUST be present in a FOCUS dataset when the provider supports prices in virtual currency and publishes unit prices exclusive of discounts.
  * PricingCurrencyListUnitPrice SHOULD be present in a FOCUS dataset when the provider supports pricing and billing in different currencies and publishes unit prices exclusive of discounts.
  * PricingCurrencyListUnitPrice may be present in all other cases.
```

> * PricingCurrencyListUnitPrice MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when the provider supports pricing and billing in different currencies and publishes unit prices exclusive of discounts. 

Author: Eliminate trailing space to fix linter.  

```suggestion
* PricingCurrencyListUnitPrice MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when the provider supports pricing and billing in different currencies and publishes unit prices exclusive of discounts.
```

# Pricing Currency List Unit Price

The Pricing Currency List Unit Price represents the suggested provider-published unit price for a single [Pricing Unit](#pricingunit) of the associated SKU, exclusive of any discounts. This price is denominated in the [Pricing Currency](#pricingcurrency). The Pricing Currency List Unit Price is commonly used for calculating savings based on various rate optimization activities.

The PricingCurrencyListUnitPrice column adheres to the following requirements:

* PricingCurrencyListUnitPrice presence in a [*FOCUS dataset*](#glossary:FOCUS-dataset) is defined as follows:
  * PricingCurrencyListUnitPrice MUST be present in a FOCUS dataset when the provider presents prices in a virtual currency (e.g. credits, tokens).

>   * PricingCurrencyListUnitPrice MUST be present in a FOCUS dataset when the provider presents prices and bills in different national currencies (e.g. priced in USD and billed in EUR).

Reviewer 2: ```suggestion
  * PricingCurrencyListUnitPrice MUST be present in a FOCUS dataset when the provider presents prices and usage in different national currencies (e.g. priced in USD and billed in EUR).
```

Author: Same point as I made [here](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/848#discussion_r1997460587).  Does this work for you, @thecloudman?

```suggestion
  * PricingCurrencyListUnitPrice MUST be present in a FOCUS dataset when the provider presents prices and usage in different national currencies (e.g. prices in USD and usage in EUR).
```

  * PricingCurrencyListUnitPrice MAY be present in a FOCUS dataset in all other cases.
* PricingCurrencyListUnitPrice MUST be of type Decimal.
* PricingCurrencyListUnitPrice MUST conform to [NumericFormat](#numericformat) requirements.
* PricingCurrencyListUnitPrice nullability is defined as follows:
  * PricingCurrencyListUnitPrice MUST be null when [ChargeCategory](#chargecategory) is "Tax".
  * PricingCurrencyListUnitPrice MUST NOT be null when ChargeCategory is "Usage" or "Purchase" and [ChargeClass](#chargeclass) is not "Correction".
  * PricingCurrencyListUnitPrice MAY be null in all other cases.
* When PricingCurrencyListUnitPrice is not null, ListUnitPrice adheres to the following additional requirements:
  * PricingCurrencyListUnitPrice MUST be a non-negative decimal value.

>   * PricingCurrencyListUnitPrice MUST be denominated in the BillingCurrency.

Reviewer 3: ```suggestion
  * PricingCurrencyListUnitPrice MUST be denominated in the PricingCurrency.
```

>   * The product of PricingCurrencyListUnitPrice and [PricingQuantity](#pricingquantity) MUST match the [ListCost](#listcost) when PricingQuantity is not null and ChargeClass is not "Correction".

Author: I am unsure the precise wording we need for revising this particular line that I lifted from `ListUnitPrice`. I would welcome a suggestion.

Author: This will follow the same convention as what @GrinsteadJ suggested [here](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/848/files/50501c7bac617f215bfcd304eb7f610a42278b9f#r2004389968) for Contracted.

```suggestion
  * The product of PricingCurrencyListUnitPrice, [PricingQuantity](#pricingquantity), and [PricingToBillingExchangeRate](#pricingtobillingexchangerate) MUST match the [ListCost](#listcost) when PricingQuantity is not null and ChargeClass is not "Correction".
```

>   * The product of PricingCurrencyListUnitPrice, [PricingQuantity](#pricingquantity), and [PricingToBillingEffectiveExchangeRate](#pricingtobillingeffectiveexchangerate) MUST match the [ListCost](#listcost) when PricingQuantity is not null and ChargeClass is not "Correction".

Author: Discussed in Mar 28 TF3.  Will strike this normative requirement in both the List and Contracted, and add supporting content for both columns to describe how to calculate the exchange rates.
